src/board_controller/inc/board_controller.h

```diff
@@ -38,6 +38,10 @@ extern "C"
     SHARED_EXPORT int CALLING_CONVENTION set_log_file (char *log_file);
     SHARED_EXPORT int CALLING_CONVENTION log_message (int log_level, char *message);
 
+    // platform methods
+    SHARED_EXPORT int CALLING_CONVENTION java_set_jnienv (
+        const struct JNINativeInterface *java_jnienv);
```

Comment:
I can not find a header for it in includes.. how does it work?

Reply:
Android and java backends often use [jni.h](https://android.googlesource.com/platform/libnativehelper/+/brillo-m9-dev/include/nativehelper/jni.h) to interoperate with java.  It's been standard for some time, but is only present in java environments.  To perform any java calls using jni.h, you need a `JNIEnv` pointer, which is a shorthand waying of saying a `struct JNINativeInterface` pointer.  The java runtime hands this pointer to all native methods.  Here I used an [elaborated type specifier](https://en.cppreference.com/w/cpp/language/elaborated_type_specifier) to declare `struct JNINative Interface` at the same time as using it in a pointer, which means jni.h is not needed to build the library.

I liked the previous approach of using a `void * platform_ptr` which could be used for other things on non-java platforms, but since you asked to only do java I thought it would be clearer to specify the type exactly.